### PR TITLE
DSD-1758: updated hex values for semantic "warning" color

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-1758](https://jira.nypl.org/browse/DSD-1758)

## This PR does the following:

- Adds the `ui.warning.tertiary` color token in order to have a light mode `"warning"` color option that meets the WCAG 2.1 AA color contrast standard.
- Updates the hex values for the `ui.warning.primary` and `ui.warning.secondary` color tokens.
- Updates the `Banner` and `StatusBadge` components to use the `ui.warning.tertiary` color for bertter accessibility in the `"warning"` variant.

## How has this been tested?

- local Storybook
- Figma, to validate color contrast ratios

## Accessibility concerns or updates

- The new `ui.warning.tertiary` color token provides a `"warning"` color that meets WCAG 2.1 color contrast requirements.

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,15 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
...
 ## Prerelease
 
+### Adds
+
+- Adds the `ui.warning.tertiary` color token in order to have a light mode `"warning"` color option that meets the WCAG 2.1 AA color contrast standard.
+
+### Updates
+
+- Updates the hex values for the `ui.warning.primary` and `ui.warning.secondary` color tokens.
+- Udpates the `Banner` and `StatusBadge` components to use the `ui.warning.tertiary` color for bertter accessibility in the `"warning"` variant.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: minor typo "Udpates"

File: src/components/StyleGuide/ColorCard.tsx [hunk 2]
```diff
@@ -469,18 +469,26 @@ export const cssVars = {
     },
     {
       name: "ui.warning.primary",
-      dataBlackColor: checkContrast(9.24),
-      dataWhiteColor: checkContrast(2.27),
-      dataBgPageColor: checkContrast(2.27),
-      dataBgDefaultColor: checkContrast(2.08),
+      dataBlackColor: checkContrast(8.53),
```
Comment (EdwinGuzman) [inline src/components/StyleGuide/ColorCard.tsx:472]: Where did these updated numbers come from?

**Review by EdwinGuzman — COMMENTED** [review]
This looks good. This is one use case where the color contrast issue came up (https://qa-research-catalog.nypl.org/research/research-catalog/search?q=spaghetti). 

Screenshot 2024-04-14 at 7 27 32 PM

This is using the warning-primary color on the white background. The new warning-primary color still fails but the tertiary color works.

Should we recommend to use the warning-tertiary color for uses such as the one that was flagged?

File: src/components/StyleGuide/ColorCard.tsx (hunk 2 above)
Comment (bigfishdesign13) [reply src/components/StyleGuide/ColorCard.tsx:472]: These are the updated contrast ratio values based on the new hex color.  I use a plugin in Figma to generate the values.

**Comment by bigfishdesign13** [comment]
> This is using the warning-primary color on the white background. The new warning-primary color still fails but the tertiary color works.
> 
> Should we recommend to use the warning-tertiary color for uses such as the one that was flagged?

@EdwinGuzman 
This is exactly why we created the new `tertiary` value.  It would have been too drastic to alter the existing `primary` and `secondary` colors to get to an acceptable contrast ratio, so we created a new one altogether to specifcially use for text.  That said, we did make slight modifications to the `primary` and `secondary` so they would aligned better with the new color.

I'd like to include some docs to say that the intention of `ui.warning.teriary` is for text, but I was not sure where to include it.

I am going to update the `ColorCard` component to include a space for `notes`.  I can add something there.

**Review by EdwinGuzman — APPROVED** [review]
The "notes" update is great. It's something we'll have to communicate to have devs look at when they're about to implement a design.

File: src/components/StyleGuide/ColorCard.tsx [hunk 3]
```diff
@@ -1320,7 +1336,19 @@ export const ColorCard = (props: PropsWithChildren<ColorCardProps>) => {
...
-              Source: {colorSource}
+              <span style={{ fontWeight: "bold" }}>Source:</span> {colorSource}
+            </Text>
+          )}
+          {notes && (
+            <Text
+              color={textColor}
+              fontWeight="medium"
+              lineHeight="20px"
+              // m="0"
```
Comment (oliviawongnyc) [inline src/components/StyleGuide/ColorCard.tsx]: Is this needed?

**Review by oliviawongnyc — APPROVED** [review]
LGTM!

File: src/components/StyleGuide/ColorCard.tsx (hunk 3 above)
Comment (bigfishdesign13) [reply src/components/StyleGuide/ColorCard.tsx]: Nope.

**Pull request merged**